pkg/task/task.go

```diff
@@ -234,6 +234,13 @@ func (m *taskExecutor) repeatWith() (err error) {
 			m.resetK8sClient(resource)
 		}
 
+		if rwExec.isTaskObjectNameRepeat() {
+			// if repetition is based on task object name itself, then the task's
+			// object name needs to be set
+			m.metaTaskExec.setObjectName(resource)
+			m.objectName = resource
```

Comment:
Is it possible to update/re-evaluate entire metatask object for every repeat?
If we decide/need to use a repeatWith resources in labelSelector then
we would have to add another logic to update those properties.

Reply:
I will try this in another attempt. I cannot guarantee this however at this point in time.

Reply:
i will try this in some other PR

Reply:
ok